src/util/optional.h

```diff
@@ -6,23 +6,26 @@
 
 #include "cereal/cereal.hpp"
 #include <memory>
+#include <optional>
+#include <type_traits>
 
 namespace stellar
 {
-template <typename T> using optional = std::shared_ptr<T>;
 
-template <typename T, class... Args>
-optional<T>
-make_optional(Args&&... args)
+template <class T> using optional = std::optional<T>;
+
+template <typename T>
+constexpr optional<T>
+make_optional(T const& arg)
```

Comment:
why do we need our own `make_optional` ?

Reply:
I'll remove this and the residual `util/optional.h` file

Reply:
thanks!